### ISSUE TITLE
PersonalDNSfilter turns off

### ISSUE THREAD

**Issue by Coasie** [IngoZenz/personaldnsfilter#299]
I have switched from Blokada 5 on our 2 phones, a Samsung S21+ and a Google Pixel 6, both running Android 14
I have made no configuration edits to PersonalDNSfilter, and have 4 of the included blocklists enabled.

I have turned off battery optimization for personalDNS filter on both phones.
S21+ > App Info > Battery > Unrestricted
Pixel 6 > App info >  App Battery Usage > Allow background usage

VPN for personalDNSfilterService is set to Always-on VPN
VPN for Blokada 5 has been turned off

personalNDSfilter turns off after some time on both phones, and the VPN key icon obviously disappears.
With the exact same setup, Blokada was never found disabled.

If I open the personalDNSfilyter app when it is closed, it opens right up and automatically enables, and the key icon appears. But it will eventually (hours or a day or two) turn off again on both phones.

Any ideas how to correct this, either on my end, or with a bug fix?

**Comment by IngoZenz** [comment]
I am also running on Pixel 6 but without any issues.
Samsung is pretty special with regards to killing background apps. See https://dontkillmyapp.com/samsung for more details.
If issue is remaining, please run pDNSf in debug mode (set debug = true in dnsfilter.conf). Once the issue occurs restart 1 time and send me the resulting logcat file from the personalDNSfilter folder.

**Comment by Coasie** [comment]
debug is now enabled, thanks.
What is the best way to send you the logcat file when ready?

**Comment by Coasie** [comment]
Just sent the S21+ logcat, the Pixel6 is still working well.

**Comment by IngoZenz** [comment]
Unfortunately not.
For the Samsung device please also check https://dontkillmyapp.com/samsung

**Comment by Coasie** [comment]
Happening on the Pixel 6 again too.

**Comment by Coasie** [comment]
This is still an issue on both Pixel 6 and S21+ several OS updates, and the issue persists.

**Comment by heini** [comment]
Same here. Motorola G85, Android 14, no modifications in pDNSf configuration.

**Comment by Coasie** [comment]
Yup, I went back to Blokada 5x which was bulletproof, and then just switched to dns.adguard-dns.com for connection settings and haven't looked back. The DNS connection option doesn't;t require any local updating or anything, it's set and forget. pDNSf was awful and it needs support which it does not have. The other solutions, work and have not needed any support.

**Comment by heini** [comment]
> Same here. Motorola G85, Android 14, no modifications in pDNSf configuration.

Could fix it by changing the background apps settings on the phone. pDNSf is running without a break since then.
